ai_oca_bridge_chatter/models/res_partner.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2025 Dixmit
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def _compute_im_status(self):
+        """
+        Override to set im_status to 'online' for partners
+        that have an associated user with an AI bridge.
+        It will be shown in general chatter as online.
+        """
+        for record in self.filtered(lambda r: r.user_ids.ai_bridge_id):
+            record.im_status = "online"
+        to_process = self.filtered(lambda r: not r.user_ids.ai_bridge_id)
+        if not to_process:
+            return
+        return super(ResPartner, to_process)._compute_im_status()
```

Comment:
This fails with singleton error when user_ids contains more than one associated user, it is not common but sometimes they do.
```suggestion
    def _compute_im_status(self):
        """
        Override to set im_status to 'online' for partners
        that have an associated user with an AI bridge.
        It will be shown in general chatter as online.
        """
        for record in self.filtered(lambda r: any(r.user_ids.mapped("is_ai_agent"))):
            record.im_status = "online"
        to_process = self.filtered(lambda r: not any(r.user_ids.mapped("is_ai_agent")))
        if not to_process:
            return
        return super(ResPartner, to_process)._compute_im_status()
```

Reply:
That happened on previous versions, but not anymore, if we are using m2o or o2m or m2m, the map works perfectly. there is no need of this.

Did you checked in a real life case?